gitlab/gitlab.go

```diff
@@ -83,13 +89,34 @@ func (hook Webhook) Parse(r *http.Request, events ...Event) (interface{}, error)
 		return nil, ErrInvalidHTTPMethod
 	}
 
+	// Add SystemHookEvents to event slice to allow parsing
+	events = append(events, SystemHookEvents)
```

Comment:
Is there are a reason we wouldn't just want the user to register this like other events they want to listen to?

P.S. the rest looks amazing!

Reply:
Not really, this is more an "batteries included" thing.
With this line you just need to provide the real event you want to parse. But i can see why someone doesn't want that. I will remove it to give the end user the choice to support systemHooks or not.